### PR TITLE
r/azurerm_linux_virtual_machine and r/azurerm_linux_virtual_machine_scale_set - secure_boot_enabled, vtpm_enabled

### LINKED ISSUE #13039: Support for full SecurityProfile for azurerm_windows_virtual_machine
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Can you please add support for adding a Security Profile block when deploying a Windows Virtual Machine. One of the Security Profile properties has already been implemented outside of a Security Profile block, however Azure has now added other attributes as part of the security profile to support Trusted Launch of Virtual Machines.  

### New or Affected Resource(s)

* azurerm_windows_virtual_machine
* azurerm_windows_virtual_machine_scale_set

### Potential Terraform Configuration

```hcl
resource "azurerm_windows_virtual_machine" "example" {
  name                = "example-machine"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  size                = "Standard_F2"
  admin_username      = "adminuser"
  admin_password      = "P@$$w0rd1234!"
  network_interface_ids = [
    azurerm_network_interface.example.id,
  ]

  os_disk {
    caching              = "ReadWrite"
    storage_account_type = "Standard_LRS"
  }

  security_profile {
    encryptionAtHost  = true
      securityType = 'TrustedLaunch'
      uefiSettings {
        secureBootEnabled = true
        vTpmEnabled = true
      }
  }

  source_image_reference {
    publisher = "MicrosoftWindowsServer"
    offer     = "WindowsServer"
    sku       = "2016-Datacenter"
    version   = "latest"
  }
}
```

### References

* https://docs.microsoft.com/en-us/rest/api/compute/virtual-machines/create-or-update#securityprofile

### PR DESCRIPTION
**Pull request by ms-henglu**
```
=== RUN   TestAccLinuxVirtualMachine_otherUefi
=== PAUSE TestAccLinuxVirtualMachine_otherUefi
=== CONT  TestAccLinuxVirtualMachine_otherUefi
--- PASS: TestAccLinuxVirtualMachine_otherUefi (863.74s)

=== RUN   TestAccLinuxVirtualMachineScaleSet_otherUefi
=== PAUSE TestAccLinuxVirtualMachineScaleSet_otherUefi
=== CONT  TestAccLinuxVirtualMachineScaleSet_otherUefi
--- PASS: TestAccLinuxVirtualMachineScaleSet_otherUefi (750.46s)


=== RUN   TestAccWindowsVirtualMachine_otherUefi
=== PAUSE TestAccWindowsVirtualMachine_otherUefi
=== CONT  TestAccWindowsVirtualMachine_otherUefi
--- PASS: TestAccWindowsVirtualMachine_otherUefi (813.99s)

=== RUN   TestAccWindowsVirtualMachineScaleSet_otherUefi
=== PAUSE TestAccWindowsVirtualMachineScaleSet_otherUefi
=== CONT  TestAccWindowsVirtualMachineScaleSet_otherUefi
--- PASS: TestAccWindowsVirtualMachineScaleSet_otherUefi (781.65s)
PASS

Process finished with exit code 0

```

Fixes #13039

### CODE REVIEW

File: internal/services/compute/linux_virtual_machine_resource_other_test.go [hunk 1]
```diff
@@ -572,6 +572,50 @@ func TestAccLinuxVirtualMachine_otherGracefulShutdownEnabled(t *testing.T) {
...
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.otherSecureBootEnabled(data, true),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
```
Comment (tombuildsstuff) [inline internal/services/compute/linux_virtual_machine_resource_other_test.go:593]: this field is ForceNew, so the update will recreate this, so we can remove this:

File: internal/services/compute/linux_virtual_machine_resource_other_test.go [hunk 2]
```diff
@@ -572,6 +572,50 @@ func TestAccLinuxVirtualMachine_otherGracefulShutdownEnabled(t *testing.T) {
 	})
 }
 
+func TestAccLinuxVirtualMachine_otherSecureBootEnabled(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_linux_virtual_machine", "test")
+	r := LinuxVirtualMachineResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.otherSecureBootEnabled(data, false),
```
Comment (tombuildsstuff) [inline internal/services/compute/linux_virtual_machine_resource_other_test.go:581]: which means this should be Enabled true:

```suggestion
			Config: r.otherSecureBootEnabled(data, true),
```

File: internal/services/compute/linux_virtual_machine_scale_set_other_resource_test.go [hunk 3]
```diff
@@ -638,6 +638,58 @@ func TestAccLinuxVirtualMachineScaleSet_otherHealthProbeUpdate(t *testing.T) {
...
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.otherVTpmEnabled(data, false),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(
+			"admin_password",
+		),
```
Comment (tombuildsstuff) [inline internal/services/compute/linux_virtual_machine_scale_set_other_resource_test.go:680]: (same here) this field is ForceNew but we're checking we can enable this field:

```suggestion
```

File: internal/services/compute/linux_virtual_machine_scale_set_other_resource_test.go [hunk 4]
```diff
@@ -2465,3 +2517,97 @@ resource "azurerm_linux_virtual_machine_scale_set" "test" {
...
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_linux_virtual_machine_scale_set" "test" {
+  name                = "acctestvmss-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  sku                 = "Standard_DS3_V2"
```
Comment (tombuildsstuff) [inline internal/services/compute/linux_virtual_machine_scale_set_other_resource_test.go:2533]: this is a machine with 4 cores and 14GB of ram.. an F1 will be more than sufficient here

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hi @ms-henglu 

Thanks for pushing those changes - I've taken a look through and left a few comments inline, if we can fix those up then this should otherwise be good to go 👍 

Thanks!

File: internal/services/compute/linux_virtual_machine_resource_other_test.go [hunk 5]
```diff
@@ -1796,3 +1840,77 @@ resource "azurerm_linux_virtual_machine" "test" {
...
+
+func (r LinuxVirtualMachineResource) otherVTpmEnabled(data acceptance.TestData, vTpmEnabled bool) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_linux_virtual_machine" "test" {
+  name                = "acctestVM-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  size                = "Standard_DS3_v2"
```
[the remark this answers is not in the document]
Comment (ms-henglu) [reply internal/services/compute/linux_virtual_machine_resource_other_test.go:1889]: I tried F1, but got "The selected VM size 'Standard_F1' cannot boot Hypervisor Generation '2'"

**Comment by ms-henglu** [comment]
Hi @tombuildsstuff ,

I've added a commit according to your suggestions, would you please take another look? thanks!

File: internal/services/compute/linux_virtual_machine_resource_other_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply internal/services/compute/linux_virtual_machine_resource_other_test.go:1875]: you'll need to use the Gen1 image too (see the other VM/VMSS tests for examples)

[the remark this answers is not in the document]
Comment (ms-henglu) [reply internal/services/compute/linux_virtual_machine_resource_other_test.go:1875]: It seems this feature can only be enabled in Gen2 image.

[the remark this answers is not in the document]
Comment (ms-henglu) [reply internal/services/compute/linux_virtual_machine_resource_other_test.go]: I changed to `Standard_B1ls`, please take another look, thanks!

**Comment by ms-henglu** [comment]
Hi @tombuildsstuff ,

I changed them to use a smaller vm size, please take another look, thanks!

**Review by tombuildsstuff — APPROVED** [review]
LGTM - thanks for making those changes @ms-henglu

**Pull request merged**